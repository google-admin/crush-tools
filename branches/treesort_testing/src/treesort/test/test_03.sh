test_number=03
description="all column sort"

subtest_desc=("AA,BB,CC,DD"
							"BB,AA,CC,DD"
							"DD,AA,CC,BB"
							"3,2,4,1 indexes"
							'AA,^BB,CC,^DD'
							'AA,^BB,^CC,DD'
							'^3,1,^4,2 indexes'
							'^3,^4,^2,^1 indexes'
							)
subtest_opts=(
'-K AA,BB,CC,DD'
'-K BB,AA,CC,DD'
'-K DD,AA,CC,BB'
'-k 3,2,4,1 -p'
'-K AA,^BB,CC,^DD'
'-K AA,^BB,^CC,DD'
'-k ^3,1,^4,2 -p'
'-k ^3,^4,^2,^1 -p'
)

for subtest in `seq 0 7`; do
  expected="$test_dir/test_$test_number.$subtest.expected"
  outfile="$test_dir/test_$test_number.$subtest.actual"

  $bin ${subtest_opts[$subtest]} \
       "$test_dir/test2.in" \
       > "$outfile"

  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" FAIL
  else
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" PASS
    rm "$outfile"
  fi
done
