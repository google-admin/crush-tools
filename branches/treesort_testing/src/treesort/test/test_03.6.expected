CCþAAþDDþBB
2009-10-29þ119þ16.2961þkX69CHRCpF
2009-10-29þ120þ16.2961þXdQde3c2LR
2009-10-29þ120þ11.9550þ5-dmH-yGym
2009-10-29þ222þ13.3454þ_Dh-EwmwHU
2009-10-29þ428þ16.2961þJw-SmuJJwH
2009-10-29þ525þ23.1071þ_mNRE6q64D
2009-10-29þ525þ21.4784þpfetgY4yYt
2009-10-29þ525þ20.0623þH3laCVXVqF
2009-10-29þ525þ16.2961þAnKSv96_yk
2009-10-29þ704þ19.8086þCwEfrzVzlN
2009-10-29þ790þ13.3454þeGWkG9dnHH
2009-10-25þ119þ21.4784þMLySuT_6L-
2009-10-25þ120þ19.8086þN1CuDuqLsq
2009-10-25þ222þ16.2961þGkijsWrtDm
2009-10-25þ379þ20.0623þi4Vn-mjUX_
2009-10-25þ692þ21.4784þK26asHqM1B
2009-10-25þ692þ21.4784þkC3oWoj1_8
2009-10-25þ704þ13.3454þgbwjbbmOR5
2009-10-25þ790þ25.4415þn1pETdd_e2
2009-10-25þ790þ21.4784þghqsjQH7CA
2009-10-25þ790þ20.0623þSEiwhaVJcv
2009-10-23þ120þ19.8086þWxgda_Rz1u
2009-10-23þ222þ13.3454þHF90x6_QiB
2009-10-23þ428þ25.4415þQg_963dGw2
2009-10-23þ692þ20.0623þgplXXJQCbD
2009-10-23þ704þ16.2961þab_ALFxdBl
2009-10-23þ876þ21.4784þ6RIZ_xP77O
2009-10-21þ119þ13.3454þD6DavOvkp5
2009-10-21þ222þ19.8086þMk8GiTvT-2
2009-10-21þ222þ11.9550þqdigsUU1EH
2009-10-21þ379þ43.1227þzHaJcq1brb
2009-10-21þ379þ20.0623þUH3YCtEsnE
2009-10-21þ428þ14.6088þQEIlvFBOAE
2009-10-21þ692þ23.1071þj-OdwoZTzm
2009-10-21þ692þ14.6088þooYbIsNctc
2009-10-21þ692þ13.3454þBuPJhoNfZY
2009-10-21þ704þ11.9550þPb0jUNXPGF
2009-10-21þ704þ11.9550þlAkjpi7B0u
2009-10-21þ790þ25.4415þ0CjQNzoprA
2009-10-21þ876þ23.1071þZ971TGdirG
2009-10-19þ119þ25.4415þNqFghFb16P
2009-10-19þ119þ20.0623þs5FX4ERZP8
2009-10-19þ120þ25.4415þ8ZDolJWFhK
2009-10-19þ120þ21.4784þZs9uH9wqwv
2009-10-19þ120þ21.4784þ_okpFO60P5
2009-10-19þ120þ19.8086þ1Qbw9NbEK2
2009-10-19þ222þ20.0623þQddzqzZhva
2009-10-19þ379þ21.4784þ-l43yW3QQX
2009-10-19þ379þ13.3454þDx5ojQn-G6
2009-10-19þ428þ19.8086þ0J2H9eMxz5
2009-10-19þ525þ43.1227þ-ca4VazCCo
2009-10-19þ692þ23.1071þuGBjnSxQwI
2009-10-19þ692þ21.4784þQOa1zDDXrc
2009-10-19þ790þ23.1071þpPrA9-ibvP
2009-10-19þ790þ21.4784þyj8FAR6sT8
2009-10-19þ790þ14.6088þQHJtym0emO
2009-10-19þ876þ21.4784þnLKFOKtBt5
2009-10-18þ119þ25.4415þczHP0Y09gO
2009-10-18þ119þ11.9550þoj-npio3Cq
2009-10-18þ120þ20.0623þQOhQYF2L4U
2009-10-18þ120þ14.6088þY0_LegXodZ
2009-10-18þ222þ20.0623þR0lHfvqeE2
2009-10-18þ222þ13.3454þdCzebYQfiH
2009-10-18þ379þ13.3454þmdg21XYvE4
2009-10-18þ428þ23.1071þi0-lOQWHXe
2009-10-18þ525þ43.1227þ-ynhLNQjXn
2009-10-18þ525þ16.2961þFJFwWbUCEg
2009-10-18þ692þ16.2961þ3Abgi5FvpK
2009-10-18þ876þ14.6088þqv6ulolvVm
2009-10-15þ119þ21.4784þ-_WgFpelwT
2009-10-15þ119þ19.8086þqHbD7IgvBs
2009-10-15þ119þ14.6088þQ99ceOGe58
2009-10-15þ120þ13.3454þbNfKeDJp35
2009-10-15þ222þ11.9550þOxNCXf_QO5
2009-10-15þ379þ43.1227þpUbcEzSLg5
2009-10-15þ379þ25.4415þccfVWBl1If
2009-10-15þ379þ20.0623þtH_urSAL6I
2009-10-15þ379þ14.6088þ1aRiJ5zvaU
2009-10-15þ704þ11.9550þV-dWH8XhzN
2009-10-15þ790þ16.2961þU_DsBKeZBO
2009-10-15þ876þ13.3454þOypnZVbJOY
2009-10-09þ120þ25.4415þOcgRMuIUVW
2009-10-09þ222þ43.1227þl0sIDKi1Mn
2009-10-09þ222þ23.1071þNrNOPtKMh2
2009-10-09þ222þ21.4784þZ_RNo9lPTh
2009-10-09þ222þ11.9550þZeP1N6ZHo8
2009-10-09þ428þ21.4784þDMqu6KjLMn
2009-10-09þ525þ19.8086þV_Srbg1Xgk
2009-10-09þ704þ21.4784þ_6GXpFrx7M
2009-10-06þ119þ43.1227þPKndexdB5Y
2009-10-06þ119þ23.1071þVjx-O1uUiy
2009-10-06þ379þ23.1071þbWuILAzu5L
2009-10-06þ428þ16.2961þw-DsqhmkP4
2009-10-06þ876þ16.2961þRHEIKKC-eN
2009-10-02þ120þ23.1071þS-BUVAIars
2009-10-02þ120þ19.8086þcOA-GflWf9
2009-10-02þ222þ43.1227þ8HPzHGZLDf
2009-10-02þ525þ19.8086þMP3BenV_PW
2009-10-02þ790þ23.1071þcEXl50cdsG
2009-10-02þ876þ16.2961þvT9F16VWxu
